riscv-c-api.md

```diff
@@ -172,5 +172,36 @@ For example:
 
 ## Intrinsic Functions
 
-Do we really have none of these?  I can't figure out
-`gcc/gcc/config/riscv/riscv-builtins.c`...
+Intrinsic functions (or intrinsics or built-ins) are functions that are expanded into instruction sequences by compilers.
+They typically provide access to functionality that is otherwise not synthesizable by compilers.
+Some intrinsics expand to different code sequenzes, depending on the available instructions from the enabled ISA extensions.
+
+Compilers typically come with their own set of architecture independent intrinsics (e.g. synchronization privimites, byte-swap, etc.).
+The RISC-V compiler backend can define additional target-specific intrinsics.
+Providing functionality via architecture independent intrinsics is the preferred method, as it improves code portability.
+
+Some intrinsics are only available if a particular header file is included.
+RISC-V header files, that enable intrinsics, require the prefix `riscv_` (e.g. `riscv_vector.h` or `riscv_crypto.h`).
+
+RISC-V intrinsics follow the following naming rule:
+
+```
+INTRINSIC ::= PREFIX MNEMONIC [ '_' SIZE ] [ '_' RET_TYPE ]
+PREFIX ::= "__rv_"
+MNEMONIC ::= Instruction name in the ISA extensions specification. Replace '.' with '_' (if any).
+SIZE ::= Bit width in case of restricted functionality (e.g. 32 or 64)
+RET_TYPE ::= (optional) Return type indication.
+```
+
+The `PREFIX` has the intention to avoid namespace collisions.
+
+```
+RISC-V intrinsics examples:
+
+orc.b rd, rs:
+type __rv_orc_b(type rs);
+int32_t __rv_orc_b_32(int32_t rs);
+
+vadd.vv vd, vs2, vs1:
+vint8m1_t __rv_vadd_vv_i8m1(vint8m1_t vs2, vint8m1_t vs1, size_t vl);
```

Comment:
Probably missing an option naming rule on "type abbreviation" (e.g. the `i` here)

Reply:
Will add. Thanks!